app/views.py

```diff
@@ -108,6 +108,31 @@ def events_list():
     return results, status.HTTP_200_OK
 
 
+@app.route("/api/events/search/", methods=['GET', 'POST'])
+def filter_or_search_events():
+    """Search or Filter the events list"""
+    page = request.args.get('page', 1, type=int)
+    category = request.args.get('category')
+    location = request.args.get('location')
+    if category:
+        filterd = Events.query.filter_by(category=category)
+    else:
+        filterd = Events.query.filter_by(location=location)
+    # response = [items for items in filterd]
+    response = []
+    
+    for results in filterd:
```

Comment:
Great implementation, of filtering. Since you're querying the db, you could also attach a `.paginate()` method on `Events.query.filter_by(location=location).paginate(page, per_page)` to specify the number of results you want displayed per page